web/client/src/views/Scan.vue

```diff
@@ -97,15 +136,43 @@ export default Vue.extend({
 
       this.$store.commit("setNumber", formatNumber(this.inputNumber));
 
-      this.scanEvent.$emit("scan");
+      try {
+        const res = await axios.post(`${config.apiUrl}/v2/numbers`, {
+          number: this.$store.state.number,
+        });
+
+        this.localData = res.data;
 
-      this.scanEvent.$on("finished", () => {
-        this.loading = false;
-      });
+        if (this.localData.valid) {
+          this.getScanners();
+          this.isLookup = true;
+        }
+      } catch (error) {
+        this.$store.commit("pushError", { message: error });
+      }
+
+      this.loading = false;
     },
     onSubmit(evt: Event) {
       evt.preventDefault();
     },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    updateInputNumber(val: any) {
+      this.inputNumber = val.e164;
+    },
+    async getScanners() {
+      try {
+        const res = await axios.get(`${config.apiUrl}/v2/scanners`);
+
+        this.scanners = res.data.scanners.filter(
+          (scanner: ScannerObject) => scanner.name !== "local"
+        );
+      } catch (error) {
+        this.$store.commit("pushError", { message: error });
+      }
+    },
```

Comment:
I think we can put this in a separate file and use it everywhere it's needed, instead of duplicating the code